### PR TITLE
Cover domain layer with unit tests

### PR DESCRIPTION
**Pull request by LordTermor**

### CODE REVIEW

**Comment by LordTermor** [comment]
Should I add a github actions job to run tests as a part of this commit or it's better to add other tests as well first?

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,6 +16,7 @@ set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++ -lc++abi")
 set(FETCHCONTENT_QUIET FALSE)
 
 option(BXT_EXPERIMENTAL_COPY_MOVE "Enable experimental copy/move operations" OFF)
+option(BXT_BUILD_TESTS "Build unit tests" OFF)
```
Comment (romangg) [inline CMakeLists.txt:19]: You can use the default option without the prefix

File: CMakeLists.txt [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ################################################################################
 cmake_minimum_required(VERSION 3.23)
 project(bxt C CXX)
+enable_testing()
```
Comment (romangg) [inline CMakeLists.txt:6]: I recommend to use CTest instead. Have you looked into it?

File: daemon/tests/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,53 @@
+################################################################################
+# Test Configuration
+################################################################################
+enable_testing()
+include(CTest)
```
Comment (romangg) [inline daemon/tests/CMakeLists.txt:5]: Here you include CTest. If you include CTest, no need to call enable_testing.

Use CTest's default BUILD_TESTING option.

**Review by romangg — COMMENTED** [review]
Looks good. I have just a few comments about the cmake setup.

Do you think we could run these tests directly in CI already on every push/PR?

**Review by romangg — APPROVED** [review]
Great work! Running in CI now too. :smiley:

**Comment by LordTermor** [comment]
It's not ideal and requires some extra build time as runs production build for tests on development environment (where all conan packages are built in debug mode) causing all deps rebuild. I can maybe run tests against debug build...

**Pull request merged**